### DISCUSSION TITLE
Simplification proposal for the language file

### DISCUSSION THREAD

**Discussion by slaweally** [forwardemail/forwardemail.net#306] [Ideas]
Friends, Isn't it more logical to use an id value to make the language file simpler? Currently, the size of the language file looks extremely large and complex. 

Let's give 6 digit random ID values now and in the future it will be easy for us to build a simpler and simpler structure. what do you think?

I would like to be personally involved in this task if it is accepted.

**Comment by titanism** [comment]
@slaweallx Hi there - we probably should switch to use LLM for translating these Markdown files and phrases instead of using Google Translate.  Our current setup uses Mandarin, which breaks apart Markdown files and phrases into chunks, ignoring sections with `notranslate` HTML class names, and a bit more.  If you'd like to dig in, look at `@ladjs/mandarin` at https://github.com/ladjs/mandarin, the files at <https://github.com/forwardemail/forwardemail.net/blob/master/jobs/translate-markdown.js> and <https://github.com/forwardemail/forwardemail.net/blob/master/jobs/translate-phrases.js> (which can be run one-off e.g. `node jobs/translate-phrases.js` currently).

**Reply by titanism** [reply]
Also it's in our build for Gulp at <https://github.com/forwardemail/forwardemail.net/blob/89aedc2f2ab2f137193ca64fdc1edb2095d030c8/gulpfile.js#L672-L685>

**Reply by titanism** [reply]
We have a set of pre-defined phrases at `config/phrases.js`, but the Markdown files from `index.md` files in `app/views` folder get converted and parsed via Mandarin and broken into chunks, which is main reason why phrases are inaccurate in some languages and the reason why the `locales/...` JSON files are so lengthy.

**Comment by slaweally** [comment]
My recommendation would be to use a language structure like the one I mentioned rather than automatic translation. I can do all the translations if necessary, but the structure we currently use makes this process extremely difficult.
